pkg/apis/crd/v1beta1/types.go

```diff
@@ -276,6 +276,87 @@ type IPPoolUsage struct {
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// IPPool defines one or multiple IP sets that can be used for flexible IPAM feature. For instance, the IPs can be
+// allocated to Pods according to IP pool specified in the Deployment annotation.
+type IPPool struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the IPPool.
+	Spec IPPoolSpec `json:"spec"`
+
+	// Most recently observed status of the pool.
+	Status IPPoolStatus `json:"status"`
+}
+
+type IPPoolSpec struct {
+	// The IP ranges of this IP pool, e.g. 10.10.0.0/24, 10.10.10.2-10.10.10.20, 10.10.10.30-10.10.10.30.
+	IPRanges []IPRange `json:"ipRanges"`
+	// The Subnet info of this IP pool. If set, all IP ranges in the IP pool should share the same subnet attributes.
```

Comment:
"If set" is confusing. Based on the rest of this PR, the `SubnetInfo` field is actually required for `IPPool`.

It seems that @tnqn suggested this comment and making this field a pointer in https://github.com/antrea-io/antrea/issues/5961#issuecomment-1926547576, so maybe he had something else in mind.

Reply:
I agree to remove `If set`. Will update.

Reply:
I just copied the comment from ExternalIPPoolSpec. `SubnetInfo` is optional in ExternalPPool. If it's not optional in IPPool, we should update the comment and make it not a pointer.

Reply:
Thanks all for the careful review. I have changed accordingly.